src/chart/XYAxis.js

```diff
@@ -359,19 +360,53 @@
             case "stagger":
                 xAxis.selectAll(".tick > text")
                     .attr("dy", function (d, i) { return 0.71 + (i % margin.overlapModulus) + "em"; })
+                    .attr("dx", 0)
                     .attr("visibility", null)
+                    .attr("transform", "rotate(0)")
                 ;
                 break;
             case "hide":
                 xAxis.selectAll(".tick > text")
                     .attr("dy", "0.71em")
+                    .attr("dx", 0)
                     .attr("visibility", function (d, i) { return i % margin.overlapModulus ? "hidden" : null; })
+                    .attr("transform", "rotate(0)")
+                ;
+                break;
+            case "rotate" :
+                var deg = -(this.xAxisLabelRotation()) || 0;
+                xAxis.selectAll(".tick > text")
+                    .each(function() {
+                        var elm = d3.select(this);
+                        if (deg !== 0 && margin.overlapModulus > 1) {
```

Comment:
This test can be done outside the "each" which will then let you conditionally fall through to the default case when modulus === 0 or deg === 0.